code/modules/mob/status_procs.dm

```diff
@@ -27,7 +27,7 @@
 			S.duration = world.time + max(amount, remaining_duration)
 		else if(amount > 0)
 			S = apply_status_effect(STATUS_EFFECT_STUN)
-			S.duration = amount
+			S.duration = world.time + amount
```

Comment:
This will not fix the problem, as it is set to world.time + duration in on_apply on the tick afterwards.